delta-app/src/main/java/io/cdap/delta/store/RemoteStateStore.java

```
@@ -0,0 +1,94 @@
+import io.cdap.delta.app.DeltaWorkerId;
+import io.cdap.delta.app.OffsetAndSequence;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Class to handle Rest Api calls to System Service [ Assessor ]
+ */
+
+public class RemoteStateStore implements StateStore {
+
+  private static final String BASE_PATH = "v1";
+  private static final String OFFSET_KEY = "offset";
+
+  private AssessmentServiceClient assessmentServiceClient;
+  private static final Gson GSON = new Gson();
+
+  public RemoteStateStore(ServiceDiscoverer context) {
+    this(new AssessmentServiceClient(context));
+  }
+
+  public RemoteStateStore(AssessmentServiceClient assessmentServiceClient) {
+    this.assessmentServiceClient = assessmentServiceClient;
+  }
+
+  @Override
+  public OffsetAndSequence readOffset(DeltaWorkerId id) {
+    String offsetAndSequenceJsonStr =
+      assessmentServiceClient.retryableApiCall(getMethodPath(id, OFFSET_KEY), HttpMethod.GET, null);
+    return GSON.fromJson(offsetAndSequenceJsonStr, OffsetAndSequence.class);
+  }
+
+  @Override
+  public void writeOffset(DeltaWorkerId id, OffsetAndSequence offset) {
+    String requestBody = GSON.toJson(offset);
+    assessmentServiceClient.retryableApiCall(getMethodPath(id, OFFSET_KEY), HttpMethod.PUT, requestBody);
+  }
+
+  @Override
+  public byte[] readState(DeltaWorkerId id, String key) {
+    String byteString = assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.GET, null);
+    return byteString.getBytes(StandardCharsets.UTF_8);
+  }
+
+  @Override
+  public void writeState(DeltaWorkerId id, String key, byte[] val) {
+    String valString = new String(val, StandardCharsets.UTF_8);
+    assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.PUT, valString);
+  }
+
+  @Override
+  public Long getLatestGeneration(String namespace, String pipelineName) {
+    return null;
```

if this and getWorkerInstances() are never expected to be called, better to throw an UnsupportedOperationException instead of returning null